plugins/source/azure/client/spec/spec.go

```diff
@@ -42,6 +42,17 @@ type Spec struct {
 	// Only accounts with many subscriptions should require modifying this setting,
 	// to either lower it to avoid network errors, or to increase it to speed up the discovery process.
 	DiscoveryConcurrency int `json:"discovery_concurrency" jsonschema:"minimum=1,default=400"`
+
+	// Retry options to pass to the Azure Go SDK, see more details here https://github.com/Azure/azure-sdk-for-go/blob/f951bf52fb68cbb978b7b95d41147693c1863366/sdk/azcore/policy/policy.go#L86
+	RetryOptions *RetryOptions `json:"retry_options"`
+}
+
+type RetryOptions struct {
+	MaxRetries           *int32 `json:"max_retries"`
+	TryTimeoutSeconds    *int   `json:"try_timeout_seconds" jsonschema:"minimum=0"`
+	RetryDelaySeconds    *int   `json:"retry_delay_seconds" jsonschema:"minimum=0"`
+	MaxRetryDelaySeconds *int   `json:"max_retry_delay_seconds" jsonschema:"minimum=0"`
+	StatusCodes          *[]int `json:"status_codes" jsonschema:"minItems=1,uniqueItems=true"`
```

Comment:
Have you validated / do we have tests to check that this json schema spec allows `nil` to be specified?

Reply:
Yes, it allows it

Reply:
Let me check about adding tests for it

Reply:
Ah, OK so there are existing tests that don't set it in https://github.com/cloudquery/cloudquery/blob/bb141c8baaa9674df82cb4d70476c0e7258f9f36/plugins/source/azure/client/spec/spec_test.go#L185 I'll add some more

Reply:
Cool, yeah I'm specifically interested in the `StatusCodes` case, because it's not clear from the schema that it allows nil to be specified, but maybe it's implied.